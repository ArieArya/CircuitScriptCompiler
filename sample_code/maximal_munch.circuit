reg orr = 1;
reg andd = 0;
wire wire123 = or(orr, andd);
wire w1 = and(andd, orr);

print(w1);
print(wire123);