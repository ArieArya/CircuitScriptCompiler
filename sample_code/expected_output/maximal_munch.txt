Lexing file: maximal_munch.circuit
Tokens:
Token.OPERATOR (Lexeme = '==')
Token.OPERATOR (Lexeme = '==')
Token.OPERATOR (Lexeme = '=')
Token.KEYWORD (Lexeme = 'wire')
Token.IDENTIFIER (Lexeme = 'wire123')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'reg')
Token.IDENTIFIER (Lexeme = 'r1')
Token.OPERATOR (Lexeme = '=')
Token.KEYWORD (Lexeme = 'and')
Token.LPAREN (Lexeme = '(')
Token.IDENTIFIER (Lexeme = 'andd')
Token.COMMA (Lexeme = ',')
Token.IDENTIFIER (Lexeme = 'orr')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
